addons/payment_stripe/models/payment.py

```diff
@@ -109,13 +109,15 @@ def _add_available_payment_method_types(self, stripe_session_data, tx_values):
         for idx, payment_method_type in enumerate(available_payment_method_types):
             stripe_session_data[f'payment_method_types[{idx}]'] = payment_method_type
 
-    def _stripe_request(self, url, data=False, method='POST'):
+    def _stripe_request(self, url, data=False, method='POST', tx=None):
```

Comment:
I'd rather pass the `idempotency_key` directly because:
1. `tx` is not passed every time
2. it's not clear that you should pass it to generate an idempotency key
3. we have more context on the API call from the method that requests it, and thus on how the key should be generated.